### PR TITLE
add rbac for openshift endpoint

### PR DESCRIPTION
**Pull request by pradeepitm12**
this handles rbac controller according to knative/pkg

Signed-off-by: Pradeep Kumar <pradkuma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: pkg/reconciler/openshift/rbac/rbac.go [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+	return &installed, err
+}
+
+func (r *Reconciler) ensureSCClusterRole(ctx context.Context) error {
+	logger := logging.FromContext(ctx)
+
+	logger.Info("finding cluster role pipeline-anyuid")
+
+	clusterRole := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{Name: "pipeline-anyuid"},
```
Comment (pradeepitm12) [inline pkg/reconciler/openshift/rbac/rbac.go:194]: will move this `"pipeline-anyuid" `to const or flag

File: pkg/reconciler/kubernetes/tektonpipeline/controller.go [hunk 2]
```diff
@@ -38,7 +38,7 @@ import (
 	"knative.dev/pkg/logging"
 )
 
-// NewController initializes the controller and is called by the generated code
+// NewController initializes the controlleramssible software and is called by the generated code
```
Comment (nikhil-thomas) [inline pkg/reconciler/kubernetes/tektonpipeline/controller.go:41]: ~~amssible software~~

File: cmd/openshift/main.go [hunk 3]
```diff
@@ -26,5 +27,6 @@ func main() {
 	sharedmain.Main("tekton-operator",
 		tektonpipeline.NewController,
 		tektontrigger.NewController,
+		rbac.NewController,
```
Comment (savitaashture) [inline cmd/openshift/main.go:30]: How this `rbac.NewController` will be called :thinking:

Comment (vdemeester) [reply cmd/openshift/main.go:30]: @savitaashture `sharedmain.Main` does that :wink:

Comment (savitaashture) [reply cmd/openshift/main.go:30]: My bad i have confused myself with invocation part :grin:

**Comment by savitaashture** [comment]
@pradeepitm12 
I tried to test locally i see some panic error in controller while calling  [roleBinding](https://github.com/tektoncd/operator/pull/168/files?file-filters%5B%5D=.go#diff-bff2086c6e5e72c24d3141397dec4fb6646b90468ebf0a5ad790260d31ef6929R231)
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1702c4e]

goroutine 839 [running]:
github.com/tektoncd/operator/pkg/reconciler/openshift/rbac.(*Reconciler).ensureSCCRoleBinding(0xc0004981c0, 0x1ea0fc0, 0xc000d3fad0, 0x0, 0x0, 0x0)
	github.com/tektoncd/operator/pkg/reconciler/openshift/rbac/rbac.go:231 +0x12e
github.com/tektoncd/operator/pkg/reconciler/openshift/rbac.(*Reconciler).ReconcileKind(0xc0004981c0, 0x1ea0fc0, 0xc000d3fad0, 0xc0011362c0, 0x1, 0xc00007f200)
	github.com/tektoncd/operator/pkg/reconciler/openshift/rbac/rbac.go:100 +0x2e5
knative.dev/pkg/client/injection/kube/reconciler/core/v1/namespace.(*reconcilerImpl).Reconcile(0xc00018f540, 0x1ea0fc0, 0xc000d3fa70, 0xc000543a30, 0x3, 0xc0003d8f48, 0x1ea0fc0)
	knative.dev/pkg@v0.0.0-20201026165741-2f75016c1368/client/injection/kube/reconciler/core/v1/namespace/reconciler.go:239 +0x1365
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc000c369c0, 0x0)
	knative.dev/pkg@v0.0.0-20201026165741-2f75016c1368/controller/controller.go:513 +0x710
knative.dev/pkg/controller.(*Impl).RunContext.func3(0xc000b831b0, 0xc000c369c0)
	knative.dev/pkg@v0.0.0-20201026165741-2f75016c1368/controller/controller.go:451 +0x53
created by knative.dev/pkg/controller.(*Impl).RunContext
	knative.dev/pkg@v0.0.0-20201026165741-2f75016c1368/controller/controller.go:449 +0x1a5

```

**Review by vdemeester — APPROVED** [review]

**Comment by pradeepitm12** [comment]
> @pradeepitm12
> I tried to test locally i see some panic error in controller while calling [roleBinding](https://github.com/tektoncd/operator/pull/168/files?file-filters%5B%5D=.go#diff-bff2086c6e5e72c24d3141397dec4fb6646b90468ebf0a5ad790260d31ef6929R231)
> 
> ```
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x1702c4e]
> 
> goroutine 839 [running]:
> github.com/tektoncd/operator/pkg/reconciler/openshift/rbac.(*Reconciler).ensureSCCRoleBinding(0xc0004981c0, 0x1ea0fc0, 0xc000d3fad0, 0x0, 0x0, 0x0)
> 	github.com/tektoncd/operator/pkg/reconciler/openshift/rbac/rbac.go:231 +0x12e
> github.com/tektoncd/operator/pkg/reconciler/openshift/rbac.(*Reconciler).ReconcileKind(0xc0004981c0, 0x1ea0fc0, 0xc000d3fad0, 0xc0011362c0, 0x1, 0xc00007f200)
> 	github.com/tektoncd/operator/pkg/reconciler/openshift/rbac/rbac.go:100 +0x2e5
> knative.dev/pkg/client/injection/kube/reconciler/core/v1/namespace.(*reconcilerImpl).Reconcile(0xc00018f540, 0x1ea0fc0, 0xc000d3fa70, 0xc000543a30, 0x3, 0xc0003d8f48, 0x1ea0fc0)
> 	knative.dev/pkg@v0.0.0-20201026165741-2f75016c1368/client/injection/kube/reconciler/core/v1/namespace/reconciler.go:239 +0x1365
> knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc000c369c0, 0x0)
> 	knative.dev/pkg@v0.0.0-20201026165741-2f75016c1368/controller/controller.go:513 +0x710
> knative.dev/pkg/controller.(*Impl).RunContext.func3(0xc000b831b0, 0xc000c369c0)
> 	knative.dev/pkg@v0.0.0-20201026165741-2f75016c1368/controller/controller.go:451 +0x53
> created by knative.dev/pkg/controller.(*Impl).RunContext
> 	knative.dev/pkg@v0.0.0-20201026165741-2f75016c1368/controller/controller.go:449 +0x1a5
> ```

I tried a few times but was unable to reproduce it. May be ones we have e2e will get it.

File: pkg/reconciler/openshift/rbac/rbac.go [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+const pipelineAnyuid = "pipeline-anyuid"
+
+// FinalizeKind removes all resources after deletion of a KnativeServing.
+func (r *Reconciler) FinalizeKind(ctx context.Context, original *v1alpha1.TektonPipeline) pkgreconciler.Event {
+	logger := logging.FromContext(ctx)
+
+	// List all KnativeServings to determine if cluster-scoped resources should be deleted.
+	kss, err := r.operatorClientSet.OperatorV1alpha1().TektonPipelines().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to list all KnativeServings: %w", err)
```
Comment (nikhil-thomas) [inline pkg/reconciler/openshift/rbac/rbac.go:50]: ~~KnativeServings~~, instead TektonPipelines

Comment (pradeepitm12) [reply pkg/reconciler/openshift/rbac/rbac.go:50]: Copy-paste error! :smile:

**Pull request merged**